### PR TITLE
Partial Search 2

### LINKED ISSUE #215: Can't use shorthand for search
If I want the comic `Uncanny X-Men Vol. 2 #211` I should be able to do e.g. `uncanny 211` or even `unca 211`

### PR DESCRIPTION
**Pull request by polygirl**
Fixes #215

### CODE REVIEW

File: src/app/filter.pipe.ts [hunk 1]
```diff
@@ -12,9 +12,14 @@ export class FilterPipe implements PipeTransform {
...
 
     searchText = searchText.toLowerCase();
+    // split the search string into an array of individual words
+    const searchTextAsWords = searchText.match(/\S+/g) || [];
 
     return items.filter( item => {
-      return item.displayText.toLowerCase().includes(searchText);
+      // results should contain all search words
+      return searchTextAsWords.every(wordItem => {
+          return item.displayText.toLowerCase().includes(wordItem);
```
Comment (SubJunk) [inline src/app/filter.pipe.ts]: Indentation

File: e2e/src/search.e2e-spec.ts [hunk 2]
```diff
@@ -16,11 +16,22 @@ describe('Search', () => {
     'should display the search div\n' +
     'should display correct search results',
   async () => {
-    await element(by.tagName('input')).sendKeys('Uncanny X-Men Vol. 1 #100');
+    await element(by.tagName('input')).sendKeys('Uncanny X-Men Vol. 1 #1');
     expect(
       await element(by.css('.search-result')).getText()
     ).toContain(
-      'Uncanny X-Men Vol. 1 #100'
+      'Uncanny X-Men Vol. 1 #1'
```
Comment (SubJunk) [inline e2e/src/search.e2e-spec.ts]: Why was this changed from 100 to 1?

Comment (polygirl) [reply e2e/src/search.e2e-spec.ts]: Intermediate step that I should have changed back.  Done now.

File: src/app/filter.pipe.ts (hunk 1 above)
Comment (polygirl) [reply src/app/filter.pipe.ts]: Fixed
